dev/src/codewind/project/Requester.ts

```diff
@@ -159,8 +162,67 @@ namespace Requester {
 
         // return doProjectRequest(project, url, body, Requester.post, "Build");
         const buildMsg = Translator.t(STRING_NS, "build");
+        if (project.connection.remote) {
+            const localPath = MCUtil.fsPathToContainerPath(project.localPath);
+            Log.i(`Clearing existing content for ${project.name} from ${project.connection.host}`);
+            await requestClear(project);
+            Log.i(`Copying updated files from ${localPath} to ${project.connection.host}`);
+            await requestUploadsRecursively(project.connection, project.id, localPath, localPath);
+        } else {
+            Log.i(`Local build from local file system at ${project.localPath}`);
+        }
         await doProjectRequest(project, ProjectEndpoints.BUILD_ACTION, body, Requester.post, buildMsg);
-        // await requestValidate(project, true);
+    }
+
+    export async function requestClear(project: Project): Promise<void> {
+        const msg = Translator.t(STRING_NS, "clear");
+        await doProjectRequest(project, ProjectEndpoints.CLEAR, {}, Requester.post, msg, true);
+    }
+
+
+    export async function requestUploadsRecursively(connection: Connection, projectId: any, inputPath: string, parentPath: string): Promise<void> {
```

Comment:
is `projectID` not a string?

Reply:
also this is the smallest of things but throughout the rest of the codebase it's projectID not projectId